### ISSUE TITLE
Detectnet for several classes

### ISSUE THREAD

**Issue by csadom** [NVIDIA/DIGITS#1619]
I saw how to train detectnet for 1 and 2 classes and i managed to train them (https://github.com/NVIDIA/DIGITS/issues/1359). Can I train 3, 4 etc multiclass detectnet classifier?  And if yes how?
Thanks

**Comment by lesterlo** [comment]
I think it is possible by modifying the .prototxt file

**Comment by shinaushin** [comment]
Do you have any tips on how to train detectnet for 2 classes?

**Comment by sulth** [comment]
Hi.I tried the network for 2 classes for Kitty dataset.It is only learning the first class of car.How can i solve it.Please help me to solve.

**Comment by elaith9** [comment]
@csadom @shinaushin @sulth I wrote two blog posts about multiclass custom object detection. Here you go:

https://labs.coria.com/blog/computer-vision/PreparingDataForCustomObjectDetectionUsingNvidiaDigits?sc_camp=EE6CEFA930EF4E91883902F5BFAC7ED9
https://labs.coria.com/en/blog/computer-vision/TrainingACustomMulticlassObjectDetectionModel?sc_camp=EE6CEFA930EF4E91883902F5BFAC7ED9

I hope it helps. Let me know if you have any questions.

**Comment by kala855** [comment]
Hi @elaith9 I was reading your posts, and a lot of posts. Please if someone already solved this problem please help me. There are very interesting. But I'm having a big problem. In fact is the same problem that you were covering in posts. I'm trying to have a two detectnet detection. However my training process shows only one of my classes with a map different from Zero the other class remains in Zero after 60 epochs of training as you can see in the image below. 

![screenshot from 2018-04-30 09-21-48](https://user-images.githubusercontent.com/1973452/39431973-f65a83ca-4c57-11e8-994e-3ad09f8597f5.png)

And the configuration on the model is this:

![image](https://user-images.githubusercontent.com/1973452/39432174-68a640e0-4c58-11e8-9c88-72ad797cbcfc.png)

[my_detectnet.txt](https://github.com/NVIDIA/DIGITS/files/1960935/my_detectnet.txt)

**Comment by dqthebt24** [comment]
Hi @kala855, did you solve the problem? 
I'm facing the same problem as you, should we use different network define?
Here is my problem, mAP of class 1 = 0
![git_1](https://user-images.githubusercontent.com/20674827/41394210-597b9860-6fd3-11e8-8dc8-23fcc931cf74.PNG)

[my_net.txt](https://github.com/NVIDIA/DIGITS/files/2100997/my_net.txt)

**Comment by elaith9** [comment]
@dqthebt24 make your current model pretrained and train again for 300-600 epochs. Success will also depend on quality of your training data and labels.

**Comment by elaith9** [comment]
@kala855 60 epochs is not enough to train your model. Consider training it for at least 600 epochs.

**Comment by dqthebt24** [comment]
@elaith9 Thank you, I did with KITTI dataset for 300 epoches. mAP of class 1 always equals to 0 through all epoches, (mAP of class 0 is #0)

**Comment by elaith9** [comment]
@dqthebt24 please read my blog posts. It took me 600 epochs with forst model which I later made pretrained and used it's weights to train for another 300 epochs.

**Comment by kala855** [comment]
@elaith9 thanks for your answer. @dqthebt24 I was trying with detectnet but I suppose that in fact I didn't use enough epochs. Im going to test again.

**Comment by MarcoGonnelli74** [comment]
I wrote a small article on how to crate a dataset and train a two class DetectNet on Digits.
You can read my post here 

https://www.deeplearning-blog.com/2020/01/31/how-to-train-a-two-class-detectnet-neural-network-on-digits/

Hope it helps 
Cheers
Marco

**Comment by ashtonaureleo** [comment]
> @dqthebt24 please read my blog posts. It took me 600 epochs with first model which I later made pretrained and used it's weights to train for another 300 epochs.

@elaith9 could you please tell me what was the learning rate for the next 300 epochs after making a pretrained model?
